packages/core/src/types/core.ts

```diff
@@ -115,4 +115,6 @@ export type PreviousStep = {
   code: string;
   /** Step execution result */
   result: any;
```

Comment:
We can make this an optional now, especially since we don't include errors under the results (moved to `error` field).